endorsement.go

```diff
@@ -0,0 +1,81 @@
+package tezosprotocol
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"golang.org/x/xerrors"
+)
+
+// Endorsement models the tezos endorsement operation type
+type Endorsement struct {
+	Level int32
+}
+
+func (e *Endorsement) String() string {
+	return fmt.Sprintf("%#v", e)
+}
+
+// GetTag implements OperationContents
+func (e *Endorsement) GetTag() ContentsTag {
+	return ContentsTagEndorsement
+}
+
+// MarshalBinary implements encoding.BinaryMarshaler
+func (e *Endorsement) MarshalBinary() ([]byte, error) {
+	buf := bytes.Buffer{}
+
+	// tag
+	buf.WriteByte(byte(e.GetTag()))
+
+	// Level
+	levelBytesBuf := new(bytes.Buffer)
+	err := binary.Write(levelBytesBuf, binary.BigEndian, e.Level)
+	if err != nil {
+		return []byte(""), err
```

Comment:
We've put in a convention of wrapping errors with `return []byte(""), xerrors.Errorf("%w", err)` anytime we catch and return an error. This, at the minimum, provides stack trace information to the error and can also be used to indicate which part of the martial / unmarshal process actually failed. I'd say even if you aren't providing more context it would be good to at least wrap it with `"%w"` to get the stack trace